core/src/cluster_slots_service/cluster_slots.rs

```diff
@@ -95,6 +95,35 @@ struct RowContent {
 }
 
 impl ClusterSlots {
+    #[inline]
+    pub fn new(root_bank: &Bank, cluster_info: &ClusterInfo) -> Self {
+        let cluster_slots = Self::default();
+        cluster_slots.update(root_bank, cluster_info);
+        cluster_slots
+    }
+
+    // Intentionally private default function to disallow uninitialized construction
+    #[inline]
+    fn default() -> Self {
+        Self {
+            cluster_slots: RwLock::new(VecDeque::new()),
+            epoch_metadata: RwLock::new(HashMap::new()),
+            current_slot: AtomicU64::default(),
+            root_epoch: RwLock::new(None),
+            cursor: Mutex::new(Cursor::default()),
+            metrics_last_report: AtomicInterval::default(),
+            metric_allocations: AtomicU64::default(),
+            metric_write_locks: AtomicU64::default(),
+        }
+    }
+
+    /// Setup cluster slots with default values only for tests.
```

Comment:
nit: I think we can drop this comment; it being DCOU + `for_tests` being in the name is fairly self-explanatory IMO